builder/tests/builder/src/main/java/io/helidon/builder/test/testsubjects/WithProviderBlueprint.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.builder.test.testsubjects;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.helidon.builder.api.Prototype;
+import io.helidon.config.metadata.Configured;
+import io.helidon.config.metadata.ConfiguredOption;
+
+@Prototype.Blueprint
+@Configured
+interface WithProviderBlueprint {
+    @ConfiguredOption(provider = true, providerType = SomeProvider.class)
+    SomeProvider.SomeService oneDiscover();
```

Comment:
for completeness, it might be good to have `SomeNonExistantService` and repeat the same decls for to ensure they behave as expected,